### ISSUE TITLE
Map transformation service and '%d' in openhab2 does not work

### ISSUE THREAD

**Issue by HolgerHees** [openhab/openhab#4965]
It seams that %d is not supported as a placeholder with map transformation in openhab2

One possible solution is to replace %d with %s but this does not solve all problems.

e.g. with a group item where my map file looks like below:

5=AN
4=Einzeln
3=Einzeln
2=Einzeln
1=Einzeln
0=AUS

in openhab 1.x, I was using %d to map the group count value

now with openhab2, I can't do this anymore. If I change my placeholder %d to %s the group value is mapped to ON/OFF which is not useable in this case.

I can solve this by using a separate STRING item which will be filled by a separate rule. But this sound like "Using a sledgehammer to crack a nut".

**Comment by 9037568** [comment]
Not sure where the source of this defect is, but it's not in OH1.

**Comment by HolgerHees** [comment]
I don't understand your answer.

if I use %d placeholder in openhab 1.x everthing works fine. But if I use the same statement in openhab 2 I get the exception "java.util.IllegalFormatConversionException"

my item definition looks like this

"Group:Switch:OR(ON, OFF)        Motiondetector_Switch           "Gartenlicht Details. [MAP(motiondetector_switch_de.map):%d]" <selfLightInfo>   (Security)"

**Comment by 9037568** [comment]
In OH2, sitemap processing is done by code from one of these repositories:
[ESH](http://github.com/eclipse/smarthome/issues)
[Core](http://github.com/openhab/openhab-core/issues)
[Distro](http://github.com/openhab/openhab-distro/issues)

This ticket will need to be opened there. I'm guessing Core, but I don't know for sure.

**Comment by kaikreuzer** [comment]
> I'm guessing Core, but I don't know for sure.

Core is usually the worst choice.
If in doubt, go for Distro, if it is clearly related to code in the org.eclipse.smarthome namespace, go for ESH.

In this situation at hand, this should go to ESH. In any case, it would be very useful if you would post log files (I assume your exception appears in the log as well?)

**Comment by HolgerHees** [comment]
there is no log entry.

I got the error message only encapsulated in the json response

and I created a new issues here https://github.com/eclipse/smarthome/issues/2818
